### ISSUE TITLE
[ENH] get_entity_vals should support BIDSPath

### ISSUE THREAD

**Issue by teonbrooks** [mne-tools/mne-bids#954]
It would be nice if `get_entity_vals` supported both a filepath or a `BIDSPath`, currently it only supports the former. Since `BIDSPath` has the necessary info, it would help simplify scripting if we could use this object.

**Comment by adam2392** [comment]
`BIDSPath` has `entities` property that can return the information as a dict

**Comment by teonbrooks** [comment]
this would be for different use case. I use `get_entity_vals` to get a list of all subjects in my study using `get_entity_vals(entity_key='subject')`. if i use the entities property, it only has the stored subject name that has been updated in `BIDSPath`. to use `get_entity_vals`, I need to provide a filepath to do this, if it could use `BIDSPath`, I can declare the path once

desired behavior
```python
bids_path = BIDSPath(root=cfg['bids_root'], session=None, task=task,
                     datatype=cfg['datatype'])
subjects_list = get_entity_vals(bids_path, entity_key='subject') 
```
where subjects has all the subjects in my study

if I use `bids_path.entities['subject']` it returns None since no information is stored there yet

**Comment by adam2392** [comment]
Ah I see. Yeah that makes sense to add. I'm in favor.

**Comment by hoechenberger** [comment]
> where subjects has all the subjects in my study

… for whom `task` and `datatype` exist, right? Or would you not want this filter applied?

**Comment by hoechenberger** [comment]
Friendly ping @teonbrooks :)

**Comment by teonbrooks** [comment]
yes, we could have it conditioned on the `task` and `datatype`, that would be useful

**Comment by teonbrooks** [comment]
I'm going to go ahead and close this. I realize that BIDSPath has `root` attribute so it's not that big of a hassle to use that instead
